src/redux/actions/embedding/loadEmbedding.js

```diff
@@ -9,10 +9,21 @@ const loadEmbedding = (experimentId, embeddingType) => async (dispatch, getState
     return null;
   }
 
-  // Do not allow loading when we are no longer in a loading state if there is no error condition.
-  if (getState().embeddings[embeddingType] && !getState().embeddings[embeddingType].error) {
-    return null;
-  }
+  // Does not load anything if experiment settings is not loaded
+  const embeddingState = getState().experimentSettings?.processing?.configureEmbedding;
+  if (!embeddingState) return null;
+
+  const embeddingConfig = Object.keys(embeddingState).reduce((conf, key) => {
+    const { method } = embeddingState[key];
+
+    // eslint-disable-next-line no-param-reassign
+    conf[key] = {
+      method,
+      methodSettings: embeddingState[key].methodSettings[method],
+    };
+
+    return conf;
+  }, {});
```

Comment:
Can you give me an example input and output for `embeddingConfig()`?

Reply:
The idea is to pass only the selected configs to the worker,

input : 
```
{
  "embeddingSettings": {
    "method": "umap",
    "methodSettings": {
      "umap": {
        "minimumDistance": 12,
        "distanceMetric": "euclidean"
      },
      "tsne": {
        "perplexity": 30,
        "learningRate": 200
      }
    }
  },
  "clusteringSettings": {
    "method": "louvain",
    "methodSettings": {
      "louvain": {
        "resolution": 0.5
      }
    }
  }
}
```

output : 
```
{
  "embeddingSettings": {
    "method": "umap",
    "methodSettings": {
      "minimumDistance": 12,
      "distanceMetric": "euclidean"
    }
  },
  "clusteringSettings": {
    "method": "louvain",
    "methodSettings": {
      "resolution": 0.5
    }
  }
}
```

Reply:
Could you add this an example to the code? Now that I'm reading it again in the morning it made sense, but it's probably better to explicitly say what's going on here.

Say, as a comment, it could be something like `Filters the embedding configuration to discard methodSettings  for currently unselected methods.`

Reply:
I change this to produce output that is suitable to the inteface requested by the API, according to the changes you requested.